src/Stack/FileDigestCache.hs

```diff
@@ -0,0 +1,24 @@
+module Stack.FileDigestCache
+  ( FileDigestCache
+  , newFileDigestCache
+  , readFileDigest
+  ) where
+
+import           Stack.Prelude
+import qualified Data.Map.Strict as Map
+import qualified Pantry.SHA256 as SHA256
+
+type FileDigestCache = IORef (Map FilePath SHA256)
+
+newFileDigestCache :: MonadIO m => m FileDigestCache
+newFileDigestCache = newIORef Map.empty
+
+readFileDigest :: MonadIO m => FileDigestCache -> FilePath -> m SHA256
+readFileDigest cache filePath = do
+  digests <- readIORef cache
+  case Map.lookup filePath digests of
+    (Just digest) -> pure digest
```

Comment:
HLint does not complain, but I wondered if the brackets around `Just digest` were required?